### PR TITLE
Support story replay

### PR DESCRIPTION
**Pull request by alanorozco**

### CODE REVIEW

File: extensions/amp-story/0.1/jsonld.js [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+ * limitations under the License.
+ */
+
+import {isJsonLdScriptTag} from '../../../src/dom';
+import {tryParseJson} from '../../../src/json';
+import {user} from '../../../src/log';
+
+const TAG = 'getJsonLd';
+
+export function getJsonLd(root) {
```
Comment (choumx) [inline extensions/amp-story/0.1/jsonld.js]: JSDoc + types please.

File: extensions/amp-story/0.1/jsonld.js [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+  const scriptTag = root.querySelector('script[type="application/ld+json"]');
+
+  if (!scriptTag || !isJsonLdScriptTag(scriptTag)) {
+    return;
```
Comment (choumx) [inline extensions/amp-story/0.1/jsonld.js]: `return null`?

File: extensions/amp-story/0.1/bookend.js [hunk 3]
```diff
@@ -209,5 +301,48 @@ export class Bookend {
...
+   * @param {!AmpDoc} ampdoc
+   * @return {{
+   *   title: string,
+   *   domainName: string,
+   *   imageUrl: (string|undefined),
+   * }}
+   * @private
+   */
+  getStoryMetadata_(ampdoc) {
+    const jsonLd = getJsonLd(this.win_.document);
```
Comment (choumx) [inline extensions/amp-story/0.1/bookend.js]: Let's use `AmpDoc#getBody()` for proper shadow DOM embed support.

File: extensions/amp-story/0.1/amp-story.js [hunk 4]
```diff
@@ -766,6 +775,13 @@ export class AmpStory extends AMP.BaseElement {
   audioStopped_() {
     this.element.classList.remove('audio-playing');
   }
+
+  /** @private */
+  replay_() {
+    this.switchTo_(
```
Comment (newmuis) [inline extensions/amp-story/0.1/amp-story.js]: Why not just

```
this.hideBookend_();
this.switchTo_(this.pages_[0].element.id);
```

File: extensions/amp-story/0.1/bookend.js [hunk 5]
```diff
@@ -79,6 +85,68 @@ function buildArticle(doc, articleData) {
...
+
+  if (opt_imageUrl) {
+    const container = createElementWithAttributes(doc, 'div', {
+      class: 'i-amphtml-story-bookend-replay-image',
+    });
+
+    const iconContainer = doc.createElement('div');
+
+    // Value is constant, so it's OK to use innerHTML here.
+    iconContainer./*OK*/innerHTML = ICONS.refresh;
```
Comment (newmuis) [inline extensions/amp-story/0.1/bookend.js]: Can you do this in CSS instead?  Ultimately, we want to move all our icons there (#11545)

File: extensions/amp-story/0.1/jsonld.js (hunk 1 above)
Comment (alanorozco) [reply extensions/amp-story/0.1/jsonld.js]: Done

File: extensions/amp-story/0.1/jsonld.js (hunk 2 above)
Comment (alanorozco) [reply extensions/amp-story/0.1/jsonld.js]: Done

File: extensions/amp-story/0.1/bookend.js (hunk 3 above)
Comment (alanorozco) [reply extensions/amp-story/0.1/bookend.js]: Done

File: extensions/amp-story/0.1/amp-story.js (hunk 4 above)
Comment (alanorozco) [reply extensions/amp-story/0.1/amp-story.js]: Done

File: extensions/amp-story/0.1/bookend.js (hunk 5 above)
Comment (alanorozco) [reply extensions/amp-story/0.1/bookend.js]: Done.

**Pull request merged**